docs/benchmarks.md

```diff
@@ -237,9 +237,34 @@ NVBENCH_BENCH_TYPES(benchmark, NVBENCH_TYPE_AXES(input_types, output_types))
 ```
 
 This would generate a total of 36 configurations and instantiate the benchmark 6
-times. Keep the rapid growth of these combinations in mind when choosing the
-number of values in an axis. See the section about combinatorial explosion for
-more examples and information.
+times.
+
+Keep the rapid growth of combinations due to multiple parameter axes in mind when
+choosing the number of values in an axis. See the section about combinatorial
+explosion for more examples and information.
+
+## Zipped/Tied Iteration of Value Axes
+
+At times multiple value axes need to be iterated like they are actually a tuple
+or zipped together. To enable this behavior you can request axes to be 'tied'
+together.
+
+```cpp
+// InputTypes: {char, int, unsigned int}
+// OutputTypes: {float, double}
+// NumInputs: {2^10, 2^20, 2^30}
+// Quality: {0.5, 1.0}
+
+using input_types = nvbench::type_list<char, int, unsigned int>;
+using output_types = nvbench::type_list<float, double>;
+NVBENCH_BENCH_TYPES(benchmark, NVBENCH_TYPE_AXES(input_types, output_types))
+  .set_type_axes_names({"InputType", "OutputType"})
+  .add_int64_axis("NumInputs", {1000, 10000, 100000, 200000, 200000, 200000})
+  .add_float64_axis("Quality", {0.05, 0.1, 0.25, 0.5, 0.75, 1.});
```

Comment:
The syntax of adding axes and then later zipping them seems like it could cause confusion, especially if you're zipping an axis multiple times. 

Can we do something like this? Or does this cause problems for type-erasure?
```
NVBENCH_BENCH_TYPES(benchmark, NVBENCH_TYPE_AXES(input_types, output_types))
  .set_type_axes_names({"InputType", "OutputType"})
  .add_zipped_axis( int64_axis{"NumInputs", {1000, 10000, 100000, 200000, 200000, 200000}}, 
                    float64_axis{"Quality", {0.05, 0.1, 0.25, 0.5, 0.75, 1.}} );
```

It'd also be nice for reusing a zipped axis multiple times to define it once:
```
auto zipped = zipped_axis{ ... };

NVBENCH_BENCH_TYPES(benchmark, NVBENCH_TYPE_AXES(input_types, output_types))
  .set_type_axes_names({"InputType", "OutputType"})
  .add_zipped_axis(zipped);
```

Reply:
I believe that adding a nice user facing api of `add_zipped_axis()` is entirely possible. I don't think `auto zipped = zipped_axis{ ... };` would be as easy as we would need type erasure for that

Reply:
I have updated the PR with `add_zipped_axis` and `add_user_iteration_axes`